include/petsc_legacy.h

```diff
@@ -53,3 +57,20 @@
 #define PetscViewerAndFormatCreate NullPetscViewerAndFormatCreate
 #define PetscViewerAndFormatDestroy PETSC_NULL_FUNCTION
 #endif
+! from petsc 3.8 onward PETSC_NULL_OBJECT is gone, a specific PETSC_NULL_XXX needs to be used
+#if PETSC_VERSION_MINOR<8
```

Comment:
Would it be better to include the major version in the test as well, just in case this is still around whenever petsc 4 happens?

Reply:
Major version testing included in #183 .